### ISSUE TITLE
2023-05: Project direction; Urgent, and Important things to do 

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#2706]
Explanation about this document:

The idea for this kind of issue is to cover a given time. Let's say we will do one every semester. Because this one is the first one, this one will last until next release. Then every semester we will close the current one and do a new one. In the next document, tasks done will be removed, new tasks may be added, importantness or urgentness may be re-evaluated, etc.

Right now, many of those issues are issues everyone waits on each others, so nothing moves forward. It happens that sometime things among the more important ones are also the ones with the least personal interest. So I encourage people to look at those issues even if that's not the topic they are naturally inclined to, otherwise some of them will never be fixed.

Things can be classified as urgent and non-urgent, important and non-important. Urgent is not something higher than Important. Those are different classifications. There can be non-urgent important things, and non-important urgent things.

Things are usually organized this way:

&nbsp;|urgent|non-urgent
-|-|-
important||
non-important||

To simplify things, I only focus on the important line for now.

----

Now the document:

## Important and Urgent

- [x] **writing:** write a blog post calling for translation contributions

- [ ] **writing:** gather the changes and start writing the next release blog post

This is both urgent and important, and one has to do it. I remind this is not an effortless task and this is a release blocker. One cannot complain the release is not done if that is not already done.

- [ ] **release:** get a release built and published within 2 months (before August 2023).

Because this will be a 3+ months release, this is only achievable if someone starts writing the blog post one or two months before (that means now, May 2023).

- [ ] **game:** https://github.com/Unvanquished/Unvanquished/issues/2700

- [x] **game:** UI for selecting the language

- [x] **game:** bad bot behavior related to skillset
This one is also considered a release blocker.

- [x] **engine:** https://github.com/DaemonEngine/Daemon/issues/798
We better want this for the next release, if we cannot have it, we should not wait that much.

- [x] **engine:** https://github.com/DaemonEngine/Daemon/issues/839 (Clang regression)
We better want this for the next release, if we cannot have it, we should not wait that much.

- [ ] **game:** https://github.com/Unvanquished/Unvanquished/issues/2682 (Clang regression)
We better want this for the next release, if we cannot have it, we should not wait that much.

About Clang regressions, we need to know if those are compiler a bugs. At least one of those bugs don't modify the variable that should be modified, so there may be many hidden bugs everywhere else if that's a compiler bug.

- We need someone able to tell us if the code is correct, and we need this one to look at the code and tell us if it is correct or not. this require some non-basic C++ skills.
- We need someone able to reduce the buggy code to a small reproducing example so we can submit it to LLVM so they can fix the compiler bug if that's a compiler bug.

## Important

Those are things we may want to make efforts at working on them, even if we are not naturally inclined to, but they are not urgent, for example we cannot block any release because of it.

- [x] **game:** flamer rework https://github.com/Unvanquished/gameplay/issues/69

- [ ] **game:** new buildpoint recovery https://github.com/Unvanquished/gameplay/issues/61

- [x] **game:** bot using ladder https://github.com/Unvanquished/Unvanquished/pull/2566

- [x] **game:** wallwalking bot https://github.com/Unvanquished/Unvanquished/pull/2694

- [x] **game:** crouching bot https://github.com/Unvanquished/Unvanquished/pull/2567

- [ ] **data:** replace the Tremulous sound files
This is among top priorities since 10 years, we are 10 years late.

- [ ] **engine:** fix the PBR texture mapping
Not fixing this prevents us to recruit texture artists, and makes us unable to use actual texture libraries like https://ambientcg.com/ , https://cc0-textures.com/ , https://publicdomaintextures.com/  and others.

- [ ] **engine:** fix or reimplement the dynamic shadows
This is a regression, even quake3 had them 20 years ago.

- [ ] **engine:** correct lighting https://github.com/DaemonEngine/Daemon/issues/270
The renderer cannot be considered modern without that. Without that, games are condemned to look like games from the 90's.

- [ ] **game:** impact on movable surface
A game not doing that looks very primitive, some other games implemented that 20 years ago!

- [ ] **engine:** fix water code https://github.com/DaemonEngine/Daemon/issues/645 https://github.com/DaemonEngine/Daemon/issues/850
This is a regression, XreaL had it. Other engines like DarkPlaces have it. Without that, games are condemned to look like games from the 90's.

**Comment by illwieckz** [comment]
I added in **Important and Urgent**:

- **game:** UI for selecting the language

One may have to create the issue if not created yet.

**Comment by illwieckz** [comment]
I added:

- **game:** https://github.com/Unvanquished/Unvanquished/issues/2700

**Comment by Viech** [comment]
> https://github.com/Unvanquished/gameplay/issues/61 this has been tested and reverted on @sweet235 but there are some things that come from it that people seems to have seen as global improvements, and I believe they could be applied to the current gameplay as well, that is: HUD that indicates how long a building will take to finish, lack of OM/RC, lack of egg/node. Those at least could be backported I think. The gameplay change itself is interesting, but after practice still need refinements.

This has been tested for roughly _one, at most two weeks_ before it was not reverted but in selected parts changed—arguably to the opposite of what I intended.¹  My follow-up version 2 adresses the bits that were deemed problematic. It was completed shortly after what you call a revert, yet it was not tested so far. I do not control the populated server and I can't enforce the test. But I do not want the version that was tested the great majority of the time to be misunderstood for what I propose and used as a basis for an assessment of https://github.com/Unvanquished/gameplay/issues/61. This is a point that I already made—more diplomatically but nevertheless very clearly—within that issue. I hope that this is now understood.

---
¹Instead of increasing build times, some structures now have strictly reduced build times. Instead of a smooth increase, the system was tuned to resemble something akin to sudden death; not sure if this is still the case.

**Comment by bmorel** [comment]
> before it was not reverted but in selected parts changed

Yes. And since then, for what I believe are other reasons (but I don't know for sure) this server is back to the old BP queue system.
And yes, I know sweet tweaks things on his server, because he tries to make players staying around. So it is not be strictly speaking your version, you're right on that. The global mechanics still has been tested, even if not the way you'd like. And as I said, I and others think there might be interesting things to do with it.
I also think the GUI improvements you bring were very much appreciated, and could also be very useful with the current BP queue system even if we keep this for just one more version. It will also make things easier to test by server owners as less commits will be involved (since part of the stuff will be already in master).
This is what I meant, sorry if that frustrated you.

**Comment by bmorel** [comment]
More importantly:

* https://github.com/Unvanquished/Unvanquished/pull/2719
* there is a draft post for the call to translations, too.

Since I'm too lazy to do `git log --oneline` (I even tried to autocomplete it by reflex!) I'll do the list when all the urgent code/gameplay problems will have been fixed. Or maybe someone else will do it, why not?

**Comment by cu-kai** [comment]
With regards to the "corpses can block movers" - this has always been the case since Tremulous.

The only current breakage I am aware of with movers is with rotating entities, where you can get stuck and sometimes even fall through `func_rotating`. You can see this quite clearly on any of these mission maps:
mission-edge (near the end of the alien obstacles)
mission-one (near the start of the alien obstacles)
mission-pierogi (near the end of the alien obstacles)

I am not currently sure if this also affects func_door_rotating.

**Comment by bmorel** [comment]
> With regards to the "corpses can block movers" - this has always been the case since Tremulous.

Ok. I think it makes little sense though, but at least that's one less regression to hunt before release (since it "never worked" I guess: why should corpses only block movers and not other entities, after all?)

**Comment by bmorel** [comment]
> The only current breakage I am aware of with movers is with rotating entities, where you can get stuck and sometimes even fall through func_rotating

bisecting suggests this change happens in ec4df1d0139e70e99b5251245c160dc697b45f8d. Reverting it on top of master tends to confirm this.
